include/util/enum.hpp

```diff
@@ -0,0 +1,34 @@
+#pragma once
+
+#include <iostream>
+#include <map>
+#include <string>
+
+namespace waybar::util {
+
+struct EnumParser {
```

Comment:
This is maybe kind of overkill :D

A simple set of if statements in a function in workspaces.cpp is fine, then everywhere else you can use the enums in switch statements.

I'll write an example when I have some more time.

Reply:
I didn't want to bloat workspaces.cpp with details of how to parse an enum.